include/eve/arch/cpu/as_register.hpp

```diff
@@ -61,91 +62,54 @@ namespace eve
 
   namespace detail
   {
-    template<typename Type, typename Cardinal> struct blob
+    template<typename Type, typename Cardinal>
+    struct  blob
     {
+      using is_product_type             = void;
       using cardinal_t                  = expected_cardinal_t<Type>;
-      static constexpr auto replication = Cardinal::value/cardinal_t::value;
-
       using value_type                  = as_wide_t<Type, cardinal_t>;
       using subvalue_type               = as_wide_t<Type, typename Cardinal::split_type>;
+      static constexpr auto replication = Cardinal::value/expected_cardinal_v<Type>;
 
-      std::array<subvalue_type,2>       segments;
+      std::array<value_type, replication> storage;
 
-      template<std::size_t I> EVE_FORCEINLINE decltype(auto) get() noexcept
+      EVE_FORCEINLINE void combine(subvalue_type const& l, subvalue_type const& h)
       {
-        if constexpr(has_aggregated_abi_v<subvalue_type>)
-        {
-          constexpr auto side = ((2*I)/replication) & 1;
-          return segments[side].storage().template get<I>();
-        }
-        else
-        {
-          constexpr auto side = I & 1;
-          return segments[side];
-        }
-      }
+        std::array<subvalue_type,2> data{l,h};
 
-      template<std::size_t I> EVE_FORCEINLINE decltype(auto) get() const noexcept
-      {
-        if constexpr(has_aggregated_abi_v<subvalue_type>)
-        {
-          constexpr auto side = ((2*I)/replication) & 1;
-          return segments[side].storage().template get<I>();
-        }
-        else
-        {
-          constexpr auto side = I & 1;
-          return segments[side];
-        }
+        auto dst = reinterpret_cast<std::byte*>(this);
+        auto src = reinterpret_cast<std::byte*>(data.data());
+        std::memcpy(dst, src, sizeof(*this));
       }
 
-      template<typename Func> void for_each(Func f)
+      EVE_FORCEINLINE auto slices() const
```

Comment:
There is a function `slice()` that should do this already, no?

Reply:
this is the function we call now in the slice(w) function when we are in aggreagetd mode. I can call it slcie() to be homogeneous. Remeber this  whoel structure is nto for public consumption and just help us maintaining the aggregation state.

Reply:
if 2 functions are doing the exact same tning they should be called the same.

Reply:
Gonna rename then